rust/src/nasl/syntax/statement.rs

```diff
@@ -536,4 +544,236 @@ mod position {
 
         assert_eq!(tests, expected.len());
     }
+
+    #[test]
+    fn primitive() {
+        let code = "1;";
+        let mut parser = parse(code);
+        let stmt = parser.next().unwrap().unwrap();
+        assert_ron_snapshot!(stmt);
+    }
```

Comment:
Please extract this into a function and use it everywhere else, i.e.

```rust
    fn snapshot(code: &str) {
        let mut parser = parse(code);
        let stmt = parser.next().unwrap().unwrap();
        assert_ron_snapshot!(stmt);
    }

    #[test]
    fn primitive() {
        snapshot("1");
    }
    ...
```

Reply:
nit: doesn't that change the line output of a failing test to the function line number? I think a macro would have been better?

Reply:
Yeah it does, so my suggestion is not the way to go. Agree that a macro would be good.